gateway/protocol.go

```diff
@@ -94,26 +114,23 @@ type ProtocolRequestContext interface {
 	// Then the observation can be:
 	//  - `maxed-out endpoint` on `endpoint_101`.
 	GetObservations() protocolobservations.Observations
-
-	// TODO_TECHDEBT(@commodity, @adshmh): Revisit all the Websocket specific functions
-	// in ProtocolRequestContext.
-	// - Too many websocket specific functions are exposed explicitly implying a poor interface.
-	// - Revisit the need for exposing these at all through a refactor?
-	//
-	// TODO_TECHDEBT(@commodity, @adshmh): Revisit casing of websocket vs Websocket vs WebSocket through.
-	ProtocolRequestContextWebsocket
 }
 
 // ProtocolRequestContextWebsocket defines the functionality expected by the gateway from the protocol,
 // specifically for websocket requests
 type ProtocolRequestContextWebsocket interface {
-	// GetWebsocketConnectionHeaders returns protocol-specific headers needed for websocket connections.
-	// These headers contain protocol-specific information like session data, service IDs, etc.
-	GetWebsocketConnectionHeaders() (http.Header, error)
-
-	// GetWebsocketEndpointURL returns the websocket URL for the selected endpoint.
-	// This URL is used to establish the websocket connection to the endpoint.
-	GetWebsocketEndpointURL() (string, error)
+	// StartWebSocketBridge creates and starts a WebSocket bridge between client and endpoint.
+	// It handles all protocol-specific setup including headers, URL generation, and connection establishment.
+	// The messageProcessor handles the actual message processing (typically the gateway's websocketRequestContext).
+	// This method sends establishment observation immediately, blocks until bridge completes, then sends closure observation.
+	StartWebSocketBridge(
```

Comment:
Looks like we can scrap this method altogether: 
- The HTTP Request and ResponseWriter should be passed to protocol when building the context. 
- Starting a bridge should already by done by protocol when building a context.
- A single observations channel (which will receive all the protocol-related observations) should be used.
- Protocol should create the observations channel, as it is the producer side of the channel.